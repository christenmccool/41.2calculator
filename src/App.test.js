import { render, screen } from '@testing-library/react';
import App from './App';
import {MemoryRouter} from 'react-router-dom';

test('renders learn react link', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

test('snapshot test', () => {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})


test('add works 2+4', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/add/2/4']}><App /></MemoryRouter>);
  expect(queryByText("6")).toBeInTheDocument();
})

test('add works -2+-4', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/add/-2/-4']}><App /></MemoryRouter>);
  expect(queryByText("-6")).toBeInTheDocument();
})

test('subtract works 4-1', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/subtract/4/1']}><App /></MemoryRouter>);
  expect(queryByText("3")).toBeInTheDocument();
})

test('subtract works -4-1', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/subtract/-4/1']}><App /></MemoryRouter>);
  expect(queryByText("-5")).toBeInTheDocument();
})


test('multiply works 2*4', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/multiply/2/4']}><App /></MemoryRouter>);
  expect(queryByText("8")).toBeInTheDocument();
})

test('multiply works 2*-4', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/multiply/2/-4']}><App /></MemoryRouter>);
  expect(queryByText("-8")).toBeInTheDocument();
})

test('subtract works 10/2', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/divide/10/2']}><App /></MemoryRouter>);
  expect(queryByText("5")).toBeInTheDocument();
})

test('subtract works -10/2', () => {
  const {queryByText} = render(<MemoryRouter initialEntries={['/divide/-10/2']}><App /></MemoryRouter>);
  expect(queryByText("-5")).toBeInTheDocument();
})