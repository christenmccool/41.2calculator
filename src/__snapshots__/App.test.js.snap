// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Not found
    </h1>
  </div>
</DocumentFragment>
`;
